### ISSUE TITLE
✨ `WithMountedWorkdir()`

### ISSUE THREAD

**Issue by kpenfound** [dagger/dagger#8726]
### What are you trying to do?

Almost every pipeline has `WithMountedDirectory().WithWorkdir()`. It would be nice to have some sugar to do this in one call with something like `WithMountedWorkdir()`

The name probably needs some more thought because it could cause confusion between `WithDirectory`/`WithMountedDirectory` vs `WithWordir` != `WithMountedWorkdir`

**Comment by kpenfound** [comment]
An optional parameter for `WithMountedDirectory` like `workdir bool` could work too. The DX in python and typescript would be fine. The DX in go would be worse than the current state though

**Comment by marcosnils** [comment]
+1 on WithMountedWorkdir

**Comment by shykes** [comment]
What would `WithMountedWorkdir` do exactly? I just use `WithMountedDirectory(".")`, doesn't that achieve the same result?

**Comment by marcosnils** [comment]
> What would `WithMountedWorkdir` do exactly? I just use `WithMountedDirectory(".")`, doesn't that achieve the same result?

It's a convenience helper for `WithMountedDirectory().WithWorkdir()` as it's generally what you end up doing in most cases.

**Comment by shykes** [comment]
I think instead of `WithMountedDirectory("/foo").WithWorkdir("/foo")` we should recommend `WithWorkdir("/foo").WithMountedDirectory(".")`. It achieves the same result I believe (avoid awkward repetition of `/foo`) without further complicating our filesystem API. I think `WithDirectory` vs `WithMountedDirectory` is already too complex for most users, adding more complexity in that area is really best avoided unless absolutely necessary IMO.

**Comment by marcosnils** [comment]
> we should recommend `WithWorkdir("/foo").WithMountedDirectory(".")`. It achieves the same result I believe

yes, this also works. 

> adding more complexity in that area is really best avoided unless absolutely necessary IMO.

I wasn't necessarily looking at this from the complexity perspective but more like a shorthand since I've found myself writing `WithMountedCache().WithWorkdir()` pretty much every time.  Pretty much the same reason I also value Golang's stdlib shorthands like `time.Since` and `time.Until`. ref: 

![Uploading image.png…]()

**Comment by kpenfound** [comment]
Yeah exactly @marcosnils , the motivation is to just type less. `WithWorkdir()` is almost always called with `WithMountedDirectory()` and this happens in almost every pipeline

**Comment by sagikazarmark** [comment]
While I understand the argument from @shykes, this does bring back some PTSD from the Docker CLI days: `-v .:/foo -w /foo`.

`WithWorkdir("/foo").WithMountedDirectory(".")` is somewhat better.

(_/me looking through all my modules to make sure I remove repetitions_)
